### ISSUE TITLE
Incorrect render datetime with formatter

### ISSUE THREAD

**Issue by s1lver** [yiisoft/yii2#17316]
### What steps will reproduce the problem?
```php
<?= GridView::widget([
    'columns' => [
        [
                'format' => [
		    'date',
		    'php:Y-m-d H:i:s',
		    'timeZone' => Date::TIMEZONE_MOSCOW
		],
		'attribute' => 'create_date'
        ]
    ]
]); ?>
```
The default formatter is UTC, but if you specify a time zone it is ignored paramert `$timeZone`
### What is the expected result?
"2019-05-22 12:33:15"
### What do you get instead?
"2019-05-22 15:33:15"

### Additional info
Formatter.php
`$this->timeZone`

> string(13) "Europe/Moscow"

` $timestamp->getTimezone()`

> object(DateTimeZone)#252 (2) { ["timezone_type"]=> int(3) ["timezone"]=> string(3) "UTC" } 

[The time zone is correct](https://www.php.net/manual/en/timezones.europe.php) 

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.18
| PHP version      | 7.3.3
| Operating system |Ubuntu

**Comment by samdark** [comment]
Timezone should override defaultTimeZone.

**Comment by alexkart** [comment]
@s1lver where do you set `$timeZone`? In the app config?

I've just checked both cases and it works correctly for me:

The first case (set timeZone on the application level):
main.php:
```php
return [
    'timeZone' => 'Europe/Moscow',
    ...
```

Second case:
```php
<?= GridView::widget([
        'dataProvider' => $dp,
        'formatter' => [
            'class' => \yii\i18n\Formatter::class,
            'timeZone' => 'Europe/Moscow',
        ],
        'columns' => [
            [
                'format' => [
                    'datetime',
                    'php:Y-m-d H:i:s'
                ],
                'attribute' => 'create_date'
            ]
        ]
    ]); ?>
```

**Comment by dicrtarasov** [comment]
When running application with configured local timeZone:
config/main.php:
```php
return [
     'timeZone' => 'Asia/Yekaterinburg'
];
```

We get incorrect time values in all GridView and other widgers, which show time from ActiveRecords models.
To fix this, we must set Fotmatter::defaultTimeZone also with Application config in main.php.
I think this is incorrect and `Formatter::defaultTimeZone` by default must be `Application::timeZone`:
```php
class Formatter {
    public function init()
    {
       if (! isset($this-defaultTimeZone)) {
             $this->defaultTimeZone = Yii::$app->timeZone;
       }
    }
}
```

**Comment by cebe** [comment]
> Timezone should override defaultTimeZone.

this is not correct. defaultTimeZone is the time zone that data is stored in (database) and timeZone is the one that is used for output, so data is converted from defaultTimeZone to timeZone by the formatter.
